### DISCUSSION TITLE
Add different rounding method

### DISCUSSION THREAD

**Discussion by yunghzz35** [josdejong/mathjs#3343] [Ideas]
There is no way to use other rounding method like banker's rounding, which is common in financial applications. Adding this feature would be great I think since this library has so many math functions that others libraries do not have. For now, I can only rely on another library like big.js. But functionalites are seems to be duplicate with this library.

**Comment by dvd101x** [comment]
You could try to extend the functionality according to your needs. Maybe something like this.

```js
function bankersRounding(value) {
  const roundedValue = math.round(value);
  const remainder = value - math.floor(value);
  if (math.equal(remainder, 0.5)) {
    return math.floor(value) % 2 === 0 ? math.floor(value) : math.ceil(value);
  }
  return roundedValue;
}

math.import({ bankersRounding : bankersRounding})
```

**Comment by josdejong** [comment]
If I'm not mistaken, the only difference with bankers rounding is rounding `0.5` to `0` instead of to `1` right?

When you configure mathjs to use BigNumbers, you can utilize the rich rounding options of the library used for the bignumbers: [decimal.js](https://mikemcl.github.io/decimal.js/#modes).

```js
math.config({ number: 'BigNumber'}) 
math.BigNumber.set({ rounding: math.BigNumber.ROUND_HALF_EVEN })

console.log(math.format(math.evaluate('round(0.5)')))      // 0
console.log(math.format(math.evaluate('round(0.225, 2)'))) // 0.22
console.log(math.format(math.evaluate('round(0.235, 2)'))) // 0.24
```

**Reply by dvd101x** [reply]
I'm not familiar with this topic, but searched online some references saying:

> numbers which are equidistant from the two nearest integers are rounded to the nearest even integer. Thus, 0.5 rounds down to 0; 1.5 rounds up to 2.

https://wiki.c2.com/?BankersRounding

**Reply by josdejong** [reply]
Ah, so then we must not use `ROUND_HALF_DOWN` but `ROUND_HALF_EVEN` I think. I'll update my example, thanks for looking this up David!

> `ROUND_HALF_EVEN` Rounds towards nearest neighbour. If equidistant, rounds towards even neighbour
> https://mikemcl.github.io/decimal.js/#modes
